src/App.js

```diff
@@ -0,0 +1,91 @@
+import { BrowserRouter as Router, Link, Redirect, Route} from 'react-router-dom'
+import TransactionsByCategory from './Components/TrabsactionsByCategory'
+import Transactions from './Components/Transactions'
+import Operations from './Components/Operations'
+import { Component } from 'react'
+import React from 'react'
+import axios from 'axios'
+import './App.css'
+
+export class App extends Component {
+
+  constructor() {
+    super()
+    this.state = {
+      data:[], 
+      balance: "",
+      redirect:null
+    }
+  }
+
+  componentDidMount = async () =>{
+    await this.getData()
+    this.calculateBalance()
+  }
+
+  getData = async () =>{
+    try{
+      let transactions = await axios.get("http://localhost:3100/transactions")
+        await this.setState({data:transactions.data})
+        this.calculateBalance()
+    }
+    catch(err){
+      console.log(err)
+    }
+  }
+  calculateBalance = () =>{
+    let balance = 0 
+    this.state.data.forEach(t => balance += parseInt(t.amount))
+    this.setState({balance})
+  }
+
+  makeTransaction = async transaction =>{
+    try{
+      await axios.post("http://localhost:3100/transactions", transaction)
+      await this.getData()
```

Comment:
You don't need to make a new request to get all data.
Just modify the `post` request to send back all the data (make `find` in the post req)